### DISCUSSION TITLE
KRE 3.1 Proposal: Are We Sure We Aren’t Trying to Fix Something That Isn’t Actually Broken?

### DISCUSSION THREAD

**Discussion by CodeWalletTeam** [kinfoundation/kin-rfcs#134] [_ Archive - KRE 3.1 Discussion]
As a developer in the Kin Ecosystem we regularly evaluate the effectiveness of the Kin Rewards Engine (KRE) and propose adjustments to the methodology when we think necessary. We feel that it is important to share our feedback as it pertains to the latest proposal. 

In our view, there are three key priorities underpinning the effectiveness of the KRE:

Demand for Kin - will the proposal result in a net increase in demand for Kin?
Fairness - is the proposal fair to all Ecosystem participants?
Simplicity - is the proposal easy for all stakeholders to understand?

We’ve mapped these 3 priorities in a table in order to compare the effectiveness of the current model (KRE 3.0) to the proposed model (KRE 3.1):

![Screen Shot 2022-02-18 at 2 35 11 PM](https://user-images.githubusercontent.com/99820695/154749985-be7b8279-4f5f-4344-a932-180f001366fc.png)

After evaluating each of the key priorities and taking into account past experiences, it is our view that the KRE 3.1 proposal would be a step backwards for the Ecosystem.
 
That said, we understand that some people in the Ecosystem might feel that “the KRE isn’t working”; that the available spends are still wanting and that the number of apps is still limited.  For some there may be a sense that “we need to do something”.

We agree. That said, our view is that this is not an issue with the KRE, but is instead a reflection of the difficult task that developers must undertake. We have seen this first hand developing Code. Despite years of progress the technology is still relatively new, resulting in unreliable transactions that lead to poor user experiences. Private keys are still difficult to manage, leading to lost funds and upset users. And the App Store rules are still complex and limiting, making it difficult to find compelling spends that Google and Apple will actually approve. 

In fact, when we talk to developers they tell us that this is the best KRE we have ever had. It drives demand, it is fair between developers, and it is simple to understand.

It might be easy to be cynical that “of course developers who get a bunch of free Kin would say this”, but the KRE has always been paying out Kin to developers, and yet this is the first version where developers are no longer complaining that it is unfair. It is also the first version that has driven net positive demand for Kin.

As a next step we recommend that the Kin Foundation reject this proposal, and instead survey Kin developers to figure out what their pain points are. Why aren’t they building more compelling spends? What is holding them back? Is it because the KRE isn’t providing the right incentives? Or is it because of something else? Once we know that we can then focus our efforts, so we can make sure we aren’t trying to fix something that isn’t actually broken.

<img width="1056" alt="Screen Shot 2022-02-18 at 2 35 58 PM" src="https://user-images.githubusercontent.com/99820695/154750085-02e2a143-8382-4074-857c-1484ff26d67f.png">

**Comment by showered** [comment]
Hi Code Team, nice to hear from you!

I think your point about simplicity is a good one. In KRE 3.0 it is clear – the more users you have, the more KRE you receive. AUB is just there to keep things fair.

Regarding developers unfairly gaming the system, you're right that there is no discussion about it (to my knowledge), but the gaming is most certainly happening. A major app in the ecosystem forces 1 Kin spends from its users every day without their knowledge or explicit consent, thus converting 100% of its users to MAS. Moreover, they are able to control their AUB by dishing out unspendable, unwithdrawable Kin to their users, and then retrieving it back via what they call "expired rewards".

This issue would to some extent be fixed in KRE 3.1, as the minimum spend would be increased to 1,000 Kin. According to the current Valid Spend Guidelines, explicit consent from the user is required when 100 Kin or more is being spent. However, as you point out, this is still very easily gamed.

To fix the above behaviour, I think we need to take some action:

- We need even more and stricter rules in the Valid Spend Guidelines.
- Members of KF should use the apps themselves semi-regularly, so they can get a feeling and understanding of how earns and spends occur. It should then become obvious which Spends don't "feel" right for the ecosystem, and then the rules can potentially be further adjusted.
- The community should be encouraged/incentivised to come forward and call out unfairness.

**Reply by ted-livingston** [reply]
I think these are good points showered. 

On the stricter spend guidelines, are spends without a user's explicit action allowed within the current rules? If so are they just not being enforced? If not should this be added and enforced?

On being unable to withdraw funds, should we consider requiring the ability to withdraw your full balance being a requirement to participate in the KRE? I know that might create complications for some developers, but perhaps we just need to bite the bullet on this one, because otherwise is it really your (the user's) Kin?

What do you think?

**Reply by showered** [reply]
*Stealth spends*

I think we have a problem right now where a couple of the top apps are taking worthless 1 Kin payments from their users simply by using the app normally. In my opinion the vast majority of users are unaware it's even happening. Therefore KRE 3.0 has become a competition of which app has the most users, and doesn't consider the quality of those Kin experiences.

In the 3.1 proposal, the minimum spend becomes 1,000 Kin. This is a really big change for a couple of reasons. Firstly, it becomes more expensive to game. If you want your 10,000 users to spend 1,000 Kin three times in a month, then you need to give them at least 30,000,000 Kin. But secondly, and most importantly, it triggers one of the valid spend rules in the KRE, which states that users must be fully aware of what they are doing, how much they are spending, and what they expect to get in return for their Kin. This will eliminate the "stealth spends" going on right now. And if the apps try and game it again? We should tweak the rules so they can't!

*Mandatory withdrawal capability*

I have always been strongly of the opinion that all apps should allow users to withdraw Kin. It's good for the ecosystem, and it levels the playing field with the other apps when it comes to defending against bots and hackers. However, having spoken to a developer recently, I'm not sure it's fair to just ban those apps from the KRE. I was told of how they had to remove their withdrawal functionality because their mainstream app with "normie" users were suddenly having their experience ruined by bots and/or "crypto people" who were only their to farm free Kin.

Maybe in a future proposal we could include "boosts" or "multipliers" to your KRE Score if you include certain ecosystem-friendly features like withdrawal. Unfortunately it would require human involvement in the calculations. It's not realistic for KF staff to keep tabs on 1,000+ apps, so the KRE formula should be fully automated by looking only at the blockchain.

**Reply by Freedom-kin** [reply]
I agree with most of what you say in your last comment.
However, the problem with gaming is much larger than we think.
AU can (and is) doubled easily.
Multiple accounts are not uncommon and every AU counts for 100,000.
Shadow wallets nobody knows of are easily kept active.

The lesson we can learn is that whatever rule you make, there will be those who game them.
Conclusion: make KRE depend on as few factors as possible.

I don't think the 1000 Kin spend rule will improve much.
If you raise the spends you can also raise the earns.
On balance not much will change.

On the problem of bots that are ruining user xp.
This is the problem when you give away money for free.
It will always attract bad actors...users as well as developers who are here just for the free KRE.

An example of KRE rules that prevent most of the problems is: 
https://github.com/kinecosystem/rewards-engine/discussions/134#discussioncomment-2297963

**Reply by Polype01** [reply]
@showered : very good points

To me, a valid spend needs to be done WILLINGLY by user's action, for a sufficient amount (1000 KIN minimum seems reasonable)

Willingly also means that if a bot is programmed to send automatically 1000 KIN to the App (gaming the KRE by increasing artificially the SPEND part), it should not be taken into account for the SPEND metrics of the KRE. 
Are there some tools to find loops as the one example from above?

Bot A sending 1000 KIN to App, -> taken into account for the Spend part of the algorithm
App sending back 1000 KIN to Bot A
Bot A sending again 1000 KIN to App -> taken into account for the Spend part of the algorithm
...

How about asking Apps to trigger a CAPCHA randomly, to ensure that the spend was actually made by a human? or install some other technical feature that proves that the spend/transaction is really made by a human?

**Reply by Freedom-kin** [reply]
Hi Polype,


I agree that a SPEND should only be valid if it is voluntary and confirmed by the user. It does not matter whether it is 1 Kin or 1000 Kin. 


About the the form of gaming you mention:

It's the **Ghost to Ghost** scam and Will described this trick earlier.
It's when an app creates hidden accounts (ghosts) in users’ phones, and sends Kin back and forth between other ghosts.

However, if the app sends 1 Kin to a user it is an EARN.
When he sends 1 Kin back it's a SPEND.

But the reward is for net SPENDS.
Net Spends are SPENDS minus EARNS.
So this (and others) will not work because 1 SPEND minus 1 EARN = zero.


The funny thing is that if they do manage to artificially increase their SPENDS, they have more sales on paper than in reality. Then the local government will say: thank you, can I get the tax on that? Therefore, you should not make the rewards  through the KRE much higher than the average tax rates. 
.

**Comment by ghost** [comment]
It would be nice to have an answer from The KF on this thread :) Both Code Team and showered brought up some good points!

**Comment by supersaltee** [comment]
Totally agree with original poster. Why fix something when it ain’t broke.

What primarily concerns us is the size of median balances in relation to others.  This is unfair to apps that don’t use Kin as their core functionality.  Not many apps are gambling apps that use Kin for wagers and as such, the nature for most apps just don’t involve the majority of their users with a significant Kin balance.  These apps will by default have a lower median balance and should not be penalized for it.   Also it seems this method discourages apps in this bucket to get more users since it will consequently bring the median down, so why bring on this conflict?

Furthermore, the reduction of the daily reward cap to devs is a downer. All devs are putting in their best effort to bring more activity to ecosystem and the last thing I’m sure everyone wants to see is a smaller pot.  I mean if kin price value was increasing that would be different story.

**Reply by skyf0xx** [reply]
Adding more users still increases the payout. The advantage of median balance is it hardly shifts with the addition of users (hence median) unless it's a significant number that was added in a short period.

That would still increase the reward since the equation is `median balance * AU^0.5`

**Comment by Freedom-kin** [comment]
Hi CodeWalletTeam,

The one question you ask is _Are We Sure We Aren't Trying to Fix Something That Isn't Actually Broken?_

It's actually a statement wrapped in a question.
The statement insinuates that the current KRE is not broken.

I tend to disagree.
Let's start with your arguments.

The table says under “Demand”:

> Since the start of 2021, we have seen a significant increase in active user balances such that the AUB approaches the cumulative KRE payments during the year.

What actually happened is that KRE payments were done to developers who in turn distributed it - mainly through staking - to AUB.
Nobody payed for those Kin. That is the absorbing you refer to.
There is no art in absorbing free Kin.

The Exhibit A you refer to says nothing (sorry...no offense )
Of course it shows a steady line in KRE payouts.
The blue line shows what users did with the holding they already had.

Around 21 Feb. 2021 Peerbet started staking with 1000%, so balances from external wallets moved to the AUB.
Aug. 26 you see a green candle when Madlipz started with a 1000% APY, so of course again Kin went from external wallets to AUB.   Several more apps joined the staking arena since.  That has **nothing** to do with absorbing any KRE.

You say this is a critical metric as it allows us to evaluate how much Kin is entering and exiting the Ecosystem. A net surplus would suggest that the KRE is operating in an economically sustainable fashion.

However, the only thing it shows is that Kin was stationary in external wallets and now it is stationary in user wallets (AUB).
Developers are rewarded for keeping the AUB as high as possible. Every spend a user makes is at the expense of the AUB (=lower KRE payout) and developers are not crazy.
So again Kin is stationary in wallets, we just call it AUB.

**Economic Activity however is the FLOW of money through an economy**.
KRE 3.0 rewards the opposite. 
So yes....KRE 3.0 is broken.

Then in the right column in your arguments against 3.1 you say:

> KRE 3.0 includes active users (AU) and active user balances (AUB) as the primary inputs. The revised proposal includes one additional metric, median spends,

This is not completely correct.
3.1 does not use the AUB any longer as a primary input at all.
It's only used to determine the caps.
Besides the AU there are two additional inputs, **Median Spend** and **Median Balance**. 

You continue with saying that based on past experience rewarding Spends has typically resulted in gaming the spend guidelines and negative demand.

This is not what happened.
All outgoing transactions where seen as Spends and therefore rewarded.
The system could not see the difference between a User to App, User to User, User to Wallet or User to Exchange transaction.
Every transaction that lowered the User Balance was considered a Spend.
That of course was massively gamed.
We were rewarding all outgoing transactions....not just Spends.
So, **as far as Spends are concerned there is no compare with past experiences.**
Now the system can see the difference between transactions.


The next part in the table is about Fairness.
You argue that KRE 3.0 is fair between developers competing for KRE and that they don't feel that others are able to unfairly game it.

However, there are several ways KRE 3.0 can be gamed, which I will not discuss here to not give anyone ideas...but I shared them with the KF.

Then how fair is it if 2 apps can get 90% of the total KRE?
That means that all other 18 apps get to share the remaining 10%.
Yes, KRE 3.0 is broken.


Simplicity
KRE 3.0 is simple, that's true but it doesn't do anything for the Kin economy.
**It hinders the flow of Kin through the economy**.
I say yes, KRE 3.0 is broken.

You say that KRE 3.0 is the first version that has driven net positive demand for Kin.
I am totally lost with this statement. Supply is obviously higher than demand and every day over 220,000,000 Kin generally was injected in the Kin economy.

The reason I'm writing this isn't because I'm a supporter of version 3.1. Absolutely not. 
The reason is that I don't want to give the impression that the current version works fine, is sufficient and doesn't need to be replaced.

We really need a better version.

**Reply by ted-livingston** [reply]
Hey Freedom. I agree with a lot of your points. Perhaps “isn’t broken” wasn't the best choice of words. Perhaps a better choice might have been “isn’t the problem”.

For example it looks like the current total ecosystem AUB is 60B Kin. At $41k per billion that’s ~$2.5mm of Kin. That means a developer with 500,000 users could double AUB if they could get each of their users to buy $5 of Kin. If they could get them to each buy $50 it would 10x AUB. And if they could do that with 50 million users they would 1000x AUB. So while I agree that speculator staking may be dominating the KRE today, I don’t think it will over time. We just need more high quality experiences.

So why aren’t we seeing more high quality experiences that get people to buy Kin?

Partly it is because it’s hard. Developers need to find an experience users will like, that Apple and Google will allow, that the technology will support, and that fits within emerging regulation. Each of these things is difficult on their own, never mind all of them at the same time. This is why the crypto industry hasn’t seen a killer app yet that my mother/brother/sister in law uses. It’s a hard problem to solve.

I think the other part is that we may need to evolve our focus away from “spends” and towards “value exchanges”. Let me explain. 

If a developer has a good spend, such as in Candy Crush or Clash of Clans, they are already making a lot of money. Why would they use Kin? Why wouldn’t they just continue to use dollars and IAP? This is further complicated by Apple and Google not allowing non IAP spends on digital goods in their app stores. I think developers will have a hard time finding compelling “spends”, and if they do, they might just switch to IAP.

Instead I think we should focus on finding experiences that are only viable because of the KRE. I think PeerBet is a good example of what a "value exchange" looks like. For the first time ever PeerBet can make money simply by getting users to exchange value with each other. They don’t need to take a transaction fee, or show ads. They just need to facilitate commerce between people in the digital world. This is what I think we (and the KRE) should be focused on finding.

So it isn’t that we shouldn’t evolve the KRE, I just think we don’t need to do “a big overhaul” again. Instead I think we need to be patient to let developers find and build compelling experiences
uniquely enabled by the KRE. In the interim perhaps we can take smaller, more focused steps to improve the KRE while still keeping it simple.

For example one developer mentioned a gaming method where a dev gives out Kin but doesn’t let a user withdraw it. Perhaps we should add a rule to the definition of an active user that says they need to be able to withdraw their Kin?

Another developer mentioned a gaming method where a developer just does 1 Kin spends. Someone had a good suggestion that perhaps we should evolve the definition of an active user to be a user who spent at least 1,000 Kin in the last 30 days?

What do you think of all this Freedom?

**Reply by Freedom-kin** [reply]
Hi Ted, thanks for responding.

You say 60B Kin is in user wallets (AUB). In that case the rest ( more than 1.6T Kin ) must be in external wallets or exchanges.

Isn't this exactly what we can expect?
Most Kin holders are investors, a lot of them from the beginning.
They never intended to be regular Kin users and are holding their Kin on ledgers etc..  


But maybe we can tempt them to spend a part in the Kin economy after all.
However, the only way that will happen is when we have good SPEND opportunities.

Same goes for regular users (non investors). They must have a reason to buy Kin. Again, it's about good SPEND opportunities.

I agree with you that it's very hard to come up with good spend opportunities. **All the reason to reward exactly that**.

Giving away money is easy and I think that shouldn't **NEED** to be rewarded.
It's the SPEND part that is hard.

Everything you reward gets bigger....and we need SPENDS.
.

**Reply by Freedom-kin** [reply]
Now please look at it from the investors perspective.
It's because of them that this project could be financed.
They invested about 100 million USD.
That's what they paid for 1 Trillion Kin.
Since then 0.7 trillion Kin has been injected in the Kin economy. KRE greatly contributed to that.

To be clear, **no one paid for that free injected Kin**. It's still going on at a rate of more than 220,000,000 Kin a day. That unpaid for Kin can be offered at exchanges. Now if investors want to sell they have to compete with all that extra (free) Kin, after all developers have bills to pay too. 
We talk about a 70% inflation in 4 years....that's huge.

How do you think investors feel?
They are not happy if they see that developers are rewarded for keeping Kin **IN** the user wallets...because that's what rewarding AUB means. Rewarding keeping money IN wallets is what **Central Banks** do if they want to **COOL DOWN** an economy. They raise interest rates and people start saving money in the bank....the spending cools down.

We need to do the opposite. We need to reward SPENDING.

Please click on **View More** to see the rest :)

**Reply by Freedom-kin** [reply]
Why is it that nobody is asking who pays for all those KRE rewards?

Who is paying for the KRE rewards?
The answers is simple....it's all the Kin holders.
They pay for it in the form of inflation.
There is not something as free money....someone always pays for it.

Inflation always leads to a lower value of the currency involved.
Looking at Kins price, I rest my case.

I think the least we can do is using the KRE to promote SPENDS.
SPENDS will fuel economic activity.
**Economic Activity is the FLOWING of money through the economy.**

That's why KRE 3.0 is broken....it hinders that flow of Kin.

**Reply by Freedom-kin** [reply]
You cite Peerbet as an example of something good.
They show how “value exchange” looks like between users.
But value exchange between users is like tipping. It's like Kin moving from your left pocket to your right pocket. It is no ECONOMIC activity.

Imagine a country where the only use of money is to pass it along to your neighbor. It's a form of activity, yes, but not an economic one.
Now imagine that the state even pays you for passing it along. You get actually paid to give money away. 

Who is paying for those rewards?
The state?
It must be extra money that is printed.
Extra money out of thin air equals inflation.

So again who is paying the rewards for “value exchange” between users?
It's all holders of Kin who pay with inflation.

We need to wake up, because we are living in a fantasy. 
The ONLY thing we can do is abandoning the AUB and **start rewarding the FLOW of KIN (SPENDS)**. 
.

**Reply by Freedom-kin** [reply]
**About the gaming opportunities and keeping the KRE simple:**

I think the more factors you consider for a KRE score, the more gaming opportunities you give.

It could be real simple.
Take the net SPENDS (spends minus earns) and reward developers with a percentage of that. 
This with a cap of lets say 50,000,000 Kin.

**e.g. net SPENDS is  1000
Reward is  1000 x 30% = 300 Kin**

That way developers have **2 sources of  income**...their own SPENDS and the 30% KRE reward. In this example the developer would have an income of 1000 plus 300 = 1300 Kin.

The fact that you deduct the EARNS from the SPENDS is an incentive for the developers to generate more SPENDS then EARNS.
SPENDS are his turnover and EARNS are costs to get it.

**More SPENDS than EARNS means that Kin starts FLOWING through the economy.**

Developers will not solely depend on KRE. They have their own turnover in the form of SPENDS.

No other factors means no other gaming possibilities.

I think this system would be a great incentive for larger apps to join the system.
 **Where do you get a 30% bonus on top of your regular net turnover?**
.

**Reply by skyf0xx** [reply]
This method has merit, but my issue with it is that the net flow is always towards the markets.

A developer gets 1,000 Kin from users (who got it from the market) and 300 Kin from the KRE. To cover costs, 1,300 is then sold in exchange for USDC. This remains constant even as the number of apps and users grow.

This is similar to KRE 1 & 2 where developers were rewarded for transactions.

Proposals that maximize user balances and P2P (no Kin back to the developer) force Kin into apps and keep it there. With a bigger economy or fewer rewards, the developer becomes a reseller of Kin to users. To illustrate:

If the KRE only gave out 1,000 Kin per day, any Kin going in to PeerBet would have to come from exchanges and would be more than the developer is getting from the KRE. Meanwhile, users would be using that Kin for P2P bets. Kin would also be more valuable so that trickle coming in from the KRE would be enough.

The same effect would happen with many more apps and users demanding more than the current 250MM/ day inflation rate.

**Reply by Freedom-kin** [reply]
### Imagine living on an island and all you can buy for USD is apples. 
Well, how many apples can you eat? How many dollars do you think I'll buy? Enough to buy apples, but I don't need thousands of dollars in my account for that. I always buy just enough dollars to buy apples.

Compare that to the neighboring island where you can buy a thousand different products for your dollars. How many dollars do you think I'm buying now?

Now it is interesting to have a stock of dollars in my account.
A whole economy has developed there and that makes it interesting for me to buy a lot of dollars.

What is the difference?
The SPEND possibilities.

**On Island 1**, the seller of apples himself will also sell the dollars he received as soon as possible, because he already has enough apples and there is nothing else he can do with dollars ( **that's the problem you described** ).

**On Island 2**, the seller can use his dollars for other products and his dollar remains in circulation without having to sell it.

### Your reasoning about the circle, exchange-user-app-exchange, is therefore only a problem on island 1.

Island 2 solved that problem by creating 1000 different SPEND possibilities. But that takes time and a policy aimed at promoting SPENDS.

That's what I'm trying to achieve with a KRE 4.0 focused on SPENDS.


But what if you try to solve the problem on island 1 by handing out free dollars? You just raise the interest on the local bank and people start saving because of the high interest (**staking?**). People don't spend their dollar on apples anymore. What little economy there is will come to a complete standstill.

**On top of that, island 1 now also suffers from enormous inflation**.
The free dollars have created no value.

**Everyone feels rich,** but they can still only buy apples that now have risen considerably in price....and they don't need more apples than they did before.

That's what will happen as a result of a KRE based on AUB. 

It's rewarding on the basis of savings (AUB).
As soon as it is actually spent, the reward for the developer disappears. So the developer would be crazy to create a lot of SPEND options for the user and the effect is that the economy remains very small and users  have no reason to hold their Kin or buy more.

### Everything you reward gets bigger....we need SPENDS.
.

**Comment by skyf0xx** [comment]
@CodeWalletTeam Thanks for this comment - I think we can have a fruitful discussion here and get to a good conclusion.

In terms of of your note on simplicity of the formula, I see your point and will attempt to address it further down. But first, you mentioned a possible gaming vector in terms of spends. When we get the Median Spend of an app, the following things happen:

1. Get the app's Active Users
2. Get total volume each user spent that day
3. Calculate the Median Spend from that.

In this way, there is no way an app could inflate the size of their spends. The only way to shift the median would be to make sure their users are spending high volumes. It's impossible to shift the median without increasing spends for > 50% of the users.

I think the only way an app could possibly do that is to create large fake spends for >50% of its Active Users, daily and that would be pretty easy to catch.

**Comment by skyf0xx** [comment]
@CodeWalletTeam Your note on simplification is very well taken. I'd like to open a discussion on this below and maybe we can come up with something better.

I'd like to start off with @tannerphilp's  [suggestion](https://github.com/kinecosystem/rewards-engine/discussions/123#discussion-3850777) to simplify it to: 

`KRE Score = Median Balance * AU`

This has some immediate advantages:
1. Apps are also recognized for AU
2. Using a Median Balance instead of the total AUB closes an attack vector of putting high balances in just a few user accounts

I have some suggestions below based on variations to this formula.


**Option 1:**
KRE Score = Median Balance * AU<sup>^0.5</sup>
We are weighting the AU here as we want it to have a smaller effect than balances (Else large apps earn disproportionately much more)

**Option 2:**
KRE Score = Median Balance * AU<sup>^0.5</sup> * Median Spend<sup>^0.1</sup>
We have added spends as well to the formula



With these simplified formulas, an app is rewarded for having the highest balances, AU and spends - depending on which one we choose above. These formulas are much simpler than the original proposal

**Reply by alexkruglov** [reply]
This is an excellent discussion. I think all participants are making valid points. My concern is primarily around gaming the system. One would think that each app should build an economy that makes sense for its users and for the ecosystem, not to game the KRE. I like the idea of rewarding larger spends, but measuring them on a daily basis feels like we have competing incentives - we want high user balances and high spends, but they are in conflict with each other. In our case, we are launching spends in the 100k-500k range and intend for users to take time to earn large enough balances to make the spends. This would be consistent with the spirit of the KRE, but in practice, we would receive low payouts because our daily AU count would be low (since it takes a while to earn enough to spend 100k). So to qualify for the KRE, we would be forced to implement an unnecessary spend (e.g., some unneeded subscription) that would take a way from the joy of using our app and not add anything meaningful to the KIN economy. A double negative.

One suggestion I would make is to make a wallet active if it conducted a spend once per month. That way, fake spends (e.g., the 1-1000 Kin unauthorized withdrawal) would not be necessary. I do like the other elements of the proposal (median rather than mean calculations, 1000 KIN spend minimums), but feel de-motivated by having to launch features that hurt our users just to qualify for KRE

**Reply by skyf0xx** [reply]
I see what you're getting at but let me clarify a bit.

The balances and spends in the equations above are coming from an app's Active Users.

So if an app has 10 AU (users who spent on that day), the equation below would get the balances and spends of those 10 users.
`KRE Score = Median Balance * AU^0.5 * Median Spend^0.1`

This isn't  much different with the current KRE which gets the total balance of the users who spent on that day. For p2p spends, the AUB generally stays high. For P to developer spends, the AUB drops a little with each spend.

This would be similar in this equation. Does that make sense?

**Comment by alexkruglov** [comment]
Yes, but I think that encourages bad behavior by developers. So what I am suggesting is that an AU is someone who has spent at least once within the last month rather than on that day. That way, developers are incentivized by real spends rather than optimizing for the metric and creating a bad user experience. In most IRL apps it just doesn't make sense for AU's to need to spend daily. To be clear, I am also implying that the current KRE is actively encouraging bad behavior by apps.

**Reply by skyf0xx** [reply]
Are you referring to the definition of AU? Currently, an AU is defined as someone who has spent at least 3x in one month. Do you mean we should redefine it to 1 spend in one month?

**Comment by alexkruglov** [comment]
Oh I may have misunderstood. I was under the impression that it’s defined as having had to spend that day. But yes, once a month would be my suggestion if goal is to maximize both savings and large(r) transactions.

**Reply by skyf0xx** [reply]
We were actually discussing this internally. With KRE 3.0, the AU is defined as 3 spends a month, mostly to filter out bots. However if the suggestion above goes through, we can also define the AU as 1 spend a month. (It's a long explanation, but bots would have very little effect with the proposal above)

**Comment by alexkruglov** [comment]
Great. I do like the idea of there being a minimum qualifying spend of 1000 Kin.

**Reply by ted-livingston** [reply]
Perhaps we should be taking smaller, more iterative steps when modifying the KRE? For example, as a small step, perhaps we should change the definition of an active user to something like "someone who has spent at least 3,000 Kin in the last 30 days". What do people think of something like that?

**Reply by skyf0xx** [reply]
Smaller steps is definitely the way to go. I am partial to changing AUB to median balances of Active Users as an initial step, since that carries the same principle of AUB, but rewards apps whose users are generally 'rich' vs apps who have a handful of whales.

Users with more Kin would naturally be able to spend more if the spend opportunities are there (Such as PeerBet and P365)

**Reply by Freedom-kin** [reply]
Hi Will,

I'm of a different opinion.
AUB is killing the FLOW of KIN through the Kin Economy.
It rewards keeping Kin IN the user wallets.
We have to abandon that system as soon as possible.

Why would we prefer to reward apps whose users are generally 'rich'?
The fact that they are rich does not mean that they SPEND a lot more Kin.
Those 'rich' users are mostly investors who are given the opportunity to earn even more Kin by staking.  That's why the AUB of some apps are higher. That's why also the Median Balance of those apps is higher. 

It's NOT because they are big spenders. It's because their Kin is STATIONARY in their user wallets.
Stationary Kin is NOT FLOWING through the economy.
It's like having money in the bank and keeping it there for the high interest.
It's money NOT SPEND.....not flowing through the economy.

Your last sentence is true.
They would be able to spend more if the spend opportunities are there.
Totally true....and that is why we have to reward SPENDS.
Everything you reward gets bigger.

**Reply by skyf0xx** [reply]
Hey Freedom, I think we're talking about a multiple optimization problem here in the AUB metric.
Rewarding on AUB will increase balances, maintain p2p (hence 'Active' User Balances) and discourage spends going to the developer. So like you mentioned, the developer isn't explicitly encouraged to increase the size of spends, just to provide spend opportunities. 

Rewarding purely on spends will reduce balances, p2p but encourage spends going to the developer (and the market). The developer will need to create something the user can buy from the developer

The question we want to answer is, what are we trying to optimize - is it *only* spends? Should we address the concern of the Kin flow from developers to the market?

**Reply by Freedom-kin** [reply]
You ask:

> what are we trying to optimize - is it _only_ spends? Should we address the concern of the Kin flow from developers to the market? 

Great question.
As you stated earlier in one of your papers, 

> To begin with the end in mind means to start with a clear understanding of your destination. You need to know where you are now so that the steps you take are in the right direction.

Where are we?  
Wel, we have an economy with 70% inflation so far.
We have very few apps ( about 20). Most activity comes from staking. 
The average SPENDS are very low.  
It seems very hard to find larger apps getting interested in the Kin economy and the price of Kin is very low.

That's sounds not very positive, but it is reality. 
It's happening despite the fact that developers are working very hard to improve their apps.

The backbone of our economy is the KRE.
We can use it to steer the economy in the direction we want.
So it's not a very hard conclusion that we are doing something wrong. 
The KRE as it is is not working as it should.
We are steering our developers in the wrong direction.

Should we be concerned about the Kin flow from developers to the market?  
The answer is Yes, but...we should not focus on that problem. It's a problem that in time automatically gets solved if you focus on SPENDS.

If a developer sells his Kin on an exchange that Kin is not 'lost'.
He can only sell it if someone else buys it for the same price.
Kin only changes hands. 

What happens if you focus on SPENDS?
At the moment, what is the incentive for a regular user (not investor) to buy Kin?  You can't do much with it.

But if there are e.g. 1000 SPEND possibilities than users will want to keep a stock of Kin in their wallet. They can actually do something with their Kin. Kin will be bought on exchanges.

Isn't that the destination we want?

So we have to focus on SPENDS. We need to ensure a FLOW of Kin through the economy. It will take some time, but as always, what you reward gets bigger. It's a proven strategy.

So far we have focused on keeping Kin in user wallets.
That is the opposite of Economic Activity.
The result is where we are today.
It would not be smart to keep doing the same and expect a different result.

We have to set our priorities.
There is a limited amount of KRE available to use. If we use too much, we will drive inflation.
So what are we gonna use it for?

I would say use it for rewarding SPENDS. It's the shortest way to our destination.

Use it only for activities that make the economy grow. P2P sounds nice, but we can't afford rewarding it with KRE. Maybe later when we have a running economy. P2P isn't really an economic activity. Wallets are also important, but we don't need 20 wallets. However, if you allow them in the KRE you can't prevent that. That's why I proposed a second fund for important, but not economic activities. Than on a case by case basis you can dicide if we fund it with public money.

**Comment by alexkruglov** [comment]
I like that definition also

**Comment by alexkruglov** [comment]
Re: withdrawing, it’s quite difficult to implement with the current Kin SDK without creating a complicated user experience. I would much rather we as an ecosystem invest in working with and adopting wallets like Phantom where the problem would solve itself.

**Reply by beeman** [reply]
It's kind of outside if this discussion, but I  think integrating an external wallet will make most user cases for most users overly complicated. Also this is not something that works well on mobile today.

So while we are actively working to have a better story for using external wallets like Phantom, (SDK-less as our first step) there are still a lot of things that don't work as expexted and we will keep our low level SDKs as they allow for a integration way simpler than is possible with any external wallets.

**Comment by alexkruglov** [comment]
Agree that’s true for now. But my assumption is that will change in the coming quarters - e.g., potentially as simple as an OpenSea/MetaMask handshake. My implication is that we should be looking to that likely reality rather than force existing apps to implement a feature users would likely find distracting and complicated.

**Reply by beeman** [reply]
There is not a single sign that this will be fixed on mobile, there is no such a thing where apps can easily deeply integrate other apps. 

Also, we don't want our ecosystem partners to force their users to install a second app which is their wallet and connect that to their app. I can't see how that story would improve usability.

**Reply by Freedom-kin** [reply]
Not sure if I fully understand what you mean. I already have an app on my mobile from my local bank and I can pay every online shop with it. Why shouldn't this be possible with crypto?

**Reply by beeman** [reply]
> I already have an app on my mobile from my local bank and I can pay every online shop with it. Why shouldn't this be possible with crypto?

That use case exists today. Use Phantom, Solflare or Trust or any other wallet and voila: you can transact KIN wherever it's supported. Nothing needs to be done on the side of KF for that use case.

However, integrating those wallet apps in the SDKs or apps in order to 'prevent apps from building a certain user interface that already exists' (which was Alex' point) is not as simple as it might seem, nor would it improve they UX for any non crypto-savvy user. 

Happy to further elaborate on why that is, but let's not do that here as there is zero relation with the KRE or this specific topic.

**Comment by alexkruglov** [comment]
Definitely happy to have that debate further in a more intimate forum, but my general belief is that it’s difficult for KF to have the technology tools that are as advanced as the full Solana ecosystem. It feels we would collectively all be best off if we had the freedom to take advantage of the latest and greatest developments in web3 tools. Today we have to live behind the limitations of what the SDK can do which slows innovation
